src/polymer/pseudo-element-scanner.ts

```diff
@@ -18,40 +18,82 @@ import {ASTNode} from 'parse5';
 import * as jsdoc from '../javascript/jsdoc';
 import {annotateElementHeader} from './docs';
 
+import {Node} from 'estree';
 import {HtmlVisitor, ParsedHtmlDocument} from '../html/html-document';
+import {Visitor} from '../javascript/estree-visitor';
+import {JavaScriptDocument} from '../javascript/javascript-document';
+import {JavaScriptScanner} from '../javascript/javascript-scanner';
 import {HtmlScanner} from '../html/html-scanner';
 import {ScannedPolymerElement} from './polymer-element';
 
+function parseComment(comment: string): ScannedPolymerElement|undefined {
+  const parsedJsdoc = jsdoc.parseJsdoc(comment);
+  const pseudoTag = jsdoc.getTag(parsedJsdoc, 'pseudoElement', 'name');
+  if (pseudoTag) {
+    let element = new ScannedPolymerElement({
+      tagName: pseudoTag,
+      jsdoc: {description: parsedJsdoc.description, tags: parsedJsdoc.tags},
+      properties: [],
+      description: parsedJsdoc.description,
+      sourceRange: undefined
+    });
+    element.pseudo = true;
+    annotateElementHeader(element);
+    return element;
+  }
+}
 /**
  * A Polymer pseudo-element is an element that is declared in an unusual way, such
  * that the analyzer couldn't normally analyze it, so instead it is declared in
  * comments.
  */
-export class PseudoElementScanner implements HtmlScanner {
-  async scan(
+export class PseudoElementScanner implements HtmlScanner, JavaScriptScanner {
+  async scanHtml(
       document: ParsedHtmlDocument,
       visit: (visitor: HtmlVisitor) => Promise<void>):
       Promise<ScannedPolymerElement[]> {
     let elements: ScannedPolymerElement[] = [];
 
     await visit((node: ASTNode) => {
       if (dom5.isCommentNode(node) && node.data && node.data.includes('@pseudoElement')) {
-        const parsedJsdoc = jsdoc.parseJsdoc(node.data);
-        const pseudoTag = jsdoc.getTag(parsedJsdoc, 'pseudoElement', 'name');
-        if (pseudoTag) {
-          let element = new ScannedPolymerElement({
-            tagName: pseudoTag,
-            jsdoc: {description: parsedJsdoc.description, tags: parsedJsdoc.tags},
-            properties: [],
-            description: parsedJsdoc.description,
-            sourceRange: document.sourceRangeForNode(node)
-          });
-          element.pseudo = true;
-          annotateElementHeader(element);
+        let element = parseComment(node.data);
+        if (element) {
+          element.sourceRange = document.sourceRangeForNode(node);
           elements.push(element);
         }
       }
     });
     return elements;
   }
+
+  async scanJs(document: JavaScriptDocument): Promise<ScannedPolymerElement[]> {
+    let elements: ScannedPolymerElement[] = [];
+
+    for (let comment of document.ast.comments) {
```

Comment:
Are these all of the comments in the entire AST, or just comments that have no other home? I could see it being both ways. Let's try testing whether this code can find a pseudoElement in an IIFE (probably not an uncommon use case). Something like:

```javascript
(() => {
  /**
   * @pseudoElement eval-element
   */
  eval("Polymer({is: 'eval-element'});");
})();
```